### DISCUSSION TITLE
Backup jails

### DISCUSSION THREAD

**Discussion by spusuf** [Jip-Hop/jailmaker#20] [Ideas]
Once jails are created a backup script would be very much appreciated, that way we can either manually run the script or schedule cron jobs to do it. Not asking for much, if the jail is contained to a single folder, then simply making a copy of the folder to a set "backups" directory with iterative naming is more than sufficient.

**Comment by aardvarkl** [comment]
Why can't you use snapshots and replication?

**Reply by Jip-Hop** [reply]
Exactly. I don't think jailmaker needs to facilitate in this regard. Besides, I don't have anything inside my jail worth backing up as all the data is stored outside the jail (and made available inside the jail using bind mounts). But if you want to backup the simplest way would be to copy the jail directories (config and rootfs) somewhere. E.g. like this: `cp -rp /mnt/mypool/jailmaker/jails/myjail /mnt/backuppool/jailmaker/jails/myjail`.

**Comment by PackElend** [comment]
> and made available inside the jail using bind mounts

so you do something like this:
>  in the task tab, make a Post Init script command "mount_nullfs /path/of/source/ /path/to/target"

**Reply by spusuf** [reply]
no, for mounting directories inside the jail add the following to the user arguments section of the jail config (just inside the jail directory): "--bind='/path/on/truenas:/path/in/jail' . For example if you had a pool named `tank` and a dataset named `vault` then you'd add ``--bind='/mnt/tank/vault:/home'`` and you would see the entirety of the contents of the vault dataset in the /home directory inside the jail. It's explained in the docs, but I'll need to edit the readme.
